lib/vsc/utils/missing.py

```diff
@@ -137,6 +137,23 @@ def star(self):
         """Return a new similar monoid."""
         return Monoid(self.null, self.mappend)
 
+class MonoidConcat(object):
+    """Variation of the Monoid object optimized to concatenate lists"""
+
+    def concatenate(self, base_list, extra_list):
+        """Extend the base list with the items of the extra list"""
+        base_list.extend(extra_list)
+        return base_list
+
+    def __call__(self, *args):
+        """When the monoid is called, concatenate all given lists"""
+        base_list = args[0]
```

Comment:
this code actually extends the first arg, instead of returning a new list with all concatted lists